## DP: Fix Makefile for compiles with modern toolchains.
## DP: Patch author: Thiemo Seufer <ths@networkno.de>

From: Thiemo Seufer <ths@networkno.de>

Fix Makefile for compiles with modern toolchains.

Signed-off-by: Thiemo Seufer <ths@networkno.de>
Signed-off-by: Martin Michlmayr <tbm@cyrius.com>

--

diff -u -p -r1.186 Makefile
--- a/arch/mips/Makefile	18 Dec 2004 01:15:52 -0000	1.186
+++ b/arch/mips/Makefile	10 Jan 2005 22:10:32 -0000
@@ -118,7 +118,7 @@ if $(CC) $$gcc_abi -S -o /dev/null -xc /
 else \
 	gcc_abi=; gcc_isa=-$(5); \
 fi; \
-gas_abi=-Wa,-$(gcc-abi); gas_cpu=$$cpu; gas_isa=-Wa,$$isa; \
+gas_abi=-Wa,-mabi=$(gcc-abi); gas_cpu=$$cpu; gas_isa=-Wa,$$isa; \
 while :; do \
 	for gas_opt in -Wa,-march= -Wa,-mcpu=; do \
 		$(CC) $$gas_abi $$gas_opt$$cpu $$gas_isa -Wa,-Z -c \
@@ -129,7 +129,7 @@ while :; do \
 	break; \
 done; \
 if test "$(gcc-abi)" != "$(gas-abi)"; then \
-	gas_abi="-Wa,-$(gas-abi) -Wa,-mgp$(gcc-abi)"; \
+	gas_abi="-Wa,-mabi=$(gas-abi) -Wa,-mgp$(gcc-abi)"; \
 fi; \
 if test "$$gcc_opt" = -march= && test -n "$$gcc_abi"; then \
 	$(CC) $$gcc_abi $$gcc_opt$$gcc_cpu -S -o /dev/null \
@@ -581,7 +597,7 @@ endif
 # will break.
 #
 core-$(CONFIG_SGI_IP32)		+= arch/mips/sgi-ip32/
-cflags-$(CONFIG_SGI_IP32)	+= -Iinclude/asm-mips/mach-ip32
+cflags-$(CONFIG_SGI_IP32)	+= -Iinclude/asm-mips/mach-ip32 -Wa,-mfix7000
 load-$(CONFIG_SGI_IP32)		+= 0xffffffff80004000
 
 #

